### PR TITLE
chore(vault): add fathom events to consent manager

### PR DESCRIPTION
**Pull request by alexcarpenter**
## "Ready for Review" checklist

- [x] The Vercel preview link has been added to this PR's description
- [ ] The Asana task has been added to this PR's description
- [ ] This PR's link has been added to the "📐 Pull Request" field on the Asana task
- [ ] This Vercel preview link has been added to the "🔍 Preview" field on the Asana task
- [x] The description template has been filled in below

---

## Relevant links

- [Preview link](https://dev-portal-8kve4fap0-hashicorp.vercel.app/) 🔎

## What

Adding `onAcceptAll` and `onManagePreferences` callbacks to vault consent manager config.

## Why

Track Fathom events on accepting and managing consent preferences.

## How

Update consent manager package, and adding callbacks to config.

## Testing

Fathom events only fire in production envs, so there isn't a solid way to detect events are firing until this code is live.

### CODE REVIEW

File: src/layouts/_proxied-dot-io/vault/index.tsx [hunk 1]
```diff
@@ -11,11 +11,14 @@ import Footer from 'components/_proxied-dot-io/vault/footer-with-props'
 import ProductSubnav from 'components/_proxied-dot-io/vault/subnav'
 import productData from 'data/vault.json'
 import query from './query.graphql'
+import { trackGoal as trackFathomGoal } from 'fathom-client'
```
Comment (dstaley) [inline src/layouts/_proxied-dot-io/vault/index.tsx:14]: @BRKalow did we ever make a generic analytics client? I feel like we talked about it, but I can't seem to track down if we ever implemented it.

Comment (alexcarpenter) [reply src/layouts/_proxied-dot-io/vault/index.tsx:14]: https://app.asana.com/0/search/1202008050032049/1201931300013818

Comment (dstaley) [reply src/layouts/_proxied-dot-io/vault/index.tsx:14]: okay great so i'm not just imagining that!

Comment (dstaley) [inline src/layouts/_proxied-dot-io/vault/index.tsx:14]: nit can you move this to line 2 so that we keep our external deps above our internal ones?

**Review by dstaley — APPROVED** [review]

**Pull request merged**